tests/e2e/analyze_graph/optimize_indexes.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2023 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+
+import pytest
+from common import connect, execute_and_fetch_all
+
+# E2E tests for checking query semantic
+# ------------------------------------
+
+
+@pytest.mark.parametrize(
+    "delete_query",
+    [
+        "ANALYZE GRAPH DELETE STATISTICS",
+        "ANALYZE GRAPH ON LABELS * DELETE STATISTICS",
+        "ANALYZE GRAPH ON LABELS :Label DELETE STATISTICS",
+        "ANALYZE GRAPH ON LABELS :Label, :NONEXISTING DELETE STATISTICS",
+    ],
+)
+def test_analyze_graph_delete_statistics(delete_query, connect):
+    """Tests that all variants of delete queries work as expected."""
+    cursor = connect.cursor()
+    execute_and_fetch_all(cursor, "FOREACH (i IN range(1, 100) | CREATE (n:Label {id1: i}));")
+    execute_and_fetch_all(cursor, "FOREACH (i IN range(1, 50) | CREATE (n:Label {id2: i % 5}));")
+    execute_and_fetch_all(cursor, "CREATE INDEX ON :Label(id1);")
+    execute_and_fetch_all(cursor, "CREATE INDEX ON :Label(id2);")
+    analyze_graph_results = execute_and_fetch_all(cursor, "ANALYZE GRAPH")
+    assert len(analyze_graph_results) == 2
+    delete_stats_results = execute_and_fetch_all(cursor, delete_query)
+    assert len(delete_stats_results) == 2
+    if delete_stats_results[0][1] == "id1":
+        first_index = 0
+    else:
+        first_index = 1
+    assert delete_stats_results[first_index][0] == "Label"
+    assert delete_stats_results[first_index][1] == "id1"
+    assert delete_stats_results[1 - first_index][0] == "Label"
+    assert delete_stats_results[1 - first_index][1] == "id2"
+    # After deleting statistics, id2 should be chosen because it has less vertices
```

Comment:
My thinking is that it might be chosen due to bug in code. It makes sense to me to run explain query before you run delete and add check that `id1` is properly chosen.